### PR TITLE
MB-11675  SC can input SIT details

### PR DESCRIPTION
**Pull request by mlm55**
## Summary
This PR makes it to where the SC can input SIT details and click "Save and Continue" if all of the required SIT fields are filled in. After saving the shipment, the SC is routed to the advance page. 

Note: 'kh-MB-12330-sc-add-full-ppm' was merged in. ~~The code changes related to this ticket are in 'ServicesCounselingEditShipmentDetails.jsx'~~ (Didn't catch that some of the work was duplicated), this PR is now just tests.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. run `DEVSEED_SUBSCENARIO=ppm_customer_flow make db_dev_e2e_populate` in the terminal
2. Access the office app
3. Login as a Service Counselor
4. Look at the PPMSC1 move and edit the PPM shipment
5. Click `Yes` for `Does the customer plan to use SIT?`
6. Click in and out of the `Estimated SIT weight`, `Estimated storage start`, and `Estimated storage end` fields without filling them in to make sure the validation errors show up.
7. Fill in the fields and click `Save and Continue`
8. Verify that the shipment was saved and you were routed to the advance page

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### CODE REVIEW

File: src/pages/Office/ServicesCounselingEditShipmentDetails/ServicesCounselingEditShipmentDetails.jsx [hunk 1]
```diff
@@ -24,6 +25,11 @@ const ServicesCounselingEditShipmentDetails = ({ match, onUpdate, isAdvancePage
   const { move, order, mtoShipments, isLoading, isError } = useEditShipmentQueries(moveCode);
   const [mutateMTOShipment] = useMutation(updateMTOShipment, {
     onSuccess: (updatedMTOShipment) => {
+      if (updatedMTOShipment.shipmentType === SHIPMENT_OPTIONS.PPM && !isAdvancePage) {
+        history.push(generatePath(servicesCounselingRoutes.SHIPMENT_ADVANCE_PATH, { moveCode, shipmentId }));
+        onUpdate('success');
+        return;
+      }
```
Comment (reggieriser) [inline src/pages/Office/ServicesCounselingEditShipmentDetails/ServicesCounselingEditShipmentDetails.jsx:32]: **question:** I noticed there's some similar routing already in the `ShipmentForm` component when the form is submitted ([reference](https://github.com/transcom/mymove/blob/72ea0312cc62c3ad0d8b752230b14de94da53a5e/src/components/Office/ShipmentForm/ShipmentForm.jsx#L200-L209)).  I put in a console.log and it looks like both are being called after the first page is submitted.  Do we need both?

**Review by reggieriser — COMMENTED** [review]
I looked through the changes that I hadn't already reviewed in [Kyle's PR](https://github.com/transcom/mymove/pull/8951) (so just `ServicesCounselingEditShipmentDetails.jsx` and the corresponding test file).  Things seem to work as expected and I verified the changes I made to SIT fields were stored in the database.  Nice job on the client tests!  Before approving, though, I did have a question about something that seems duplicated (but maybe I'm missing some context).

File: src/pages/Office/ServicesCounselingEditShipmentDetails/ServicesCounselingEditShipmentDetails.jsx (hunk 1 above)
Comment (mlm55) [reply src/pages/Office/ServicesCounselingEditShipmentDetails/ServicesCounselingEditShipmentDetails.jsx]: ah, good catch. I merged that branch later in the process, and didn't catch that duplication. i'll remove it!

**Review by reggieriser — APPROVED** [review]
Thanks for removing the duplication!  LGTM.  (I don't think Kyle's PR is merged yet, so we'll need to coordinate.)

**Pull request merged**